File: scripts/export-copy-to-docx.mjs
#!/usr/bin/env node
/**
 * Export public-site English copy into one .docx per page.
 * Run: node scripts/export-copy-to-docx.mjs
 */

import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";
import {
  Document,
  Packer,
  Paragraph,
  TextRun,
  HeadingLevel,
} from "docx";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.resolve(__dirname, "..");
const SRC = path.join(ROOT, "src");
const OUT_DIR = path.join(ROOT, "suhucustom-文案导出");

const SERVICE_GROUPS = [
  {
    titleEn: "Tops & Activewear",
    items: [
      { slug: "t-shirts", nameEn: "T-shirts" },
      { slug: "hoodies-sweatshirts", nameEn: "Hoodies & Sweatshirts" },
      { slug: "activewear-athleisure", nameEn: "Activewear & Athleisure" },
      { slug: "gym-sportswear", nameEn: "Gym & Sportswear" },
    ],
  },
  {
    titleEn: "Bottoms & Underwear",
    items: [
      { slug: "leggings", nameEn: "Leggings" },
      { slug: "jeans-denim", nameEn: "Jeans & Denim" },
      { slug: "underwear-bras", nameEn: "Underwear & Bras" },
      { slug: "swimwear", nameEn: "Swimwear" },
    ],
  },
  {
    titleEn: "Accessories & Headwear",
    items: [
      { slug: "hats-headwear", nameEn: "Hats & Headwear" },
      { slug: "socks", nameEn: "Socks" },
      { slug: "neck-gaiters", nameEn: "Neck Gaiters" },
      { slug: "leather-goods", nameEn: "Leather Goods" },
    ],
  },
  {
    titleEn: "Specialized & Home",
    items: [
      { slug: "uniforms", nameEn: "Uniforms" },
      { slug: "baby-kids-clothing", nameEn: "Baby & Kids Clothing" },
      { slug: "towels", nameEn: "Towels" },
      { slug: "cushion-covers", nameEn: "Cushion Covers" },
    ],
  },
];

const CUSTOMIZATION_ITEMS = [
  { slug: "printing", nameEn: "Printing" },
  { slug: "embroidery", nameEn: "Embroidery" },
  { slug: "private-label", nameEn: "Private Label" },
  { slug: "tech-pack-design", nameEn: "Tech Pack Design" },
];

const SERVICE_COMPONENT_DIRS = {
  "t-shirts": "components/services/tshirts",
  "hoodies-sweatshirts": "components/services/hoodies",
  "activewear-athleisure": "components/services/activewear",
  "gym-sportswear": "components/services/gym",
  leggings: "components/services/leggings",
  "jeans-denim": "components/services/jeans",
  socks: "components/services/socks",
  "neck-gaiters": "components/services/neck-gaiters",
  "leather-goods": "components/services/leather-goods",
  towels: "components/services/towels",
  "cushion-covers": "components/services/cushion-covers",
};

const SKIP_RE =
  /(^\/|https?:\/\/|@\/|\.\/|#|\.tsx?$|\.svg|\.png|YOUR_|wa\.me|placeholder|className|href=|src=|object-|flex-|max-w-|aria-|viewBox)/i;
const TAILWIND_LIKE = /^[a-z0-9]+(?:-[a-z0-9]+)+$/;
const HAS_CHINESE = /[\u4e00-\u9fff]/;

function findService(slug) {
  for (const group of SERVICE_GROUPS) {
    const item = group.items.find((i) => i.slug === slug);
    if (item) return { item, group };
  }
  return null;
}

function decodeJsString(raw) {
  let s = raw.slice(1, -1);
  s = s.replace(/\\n/g, "\n").replace(/\\"/g, '"').replace(/\\'/g, "'");
  return s.replace(/\s+/g, " ").trim();
}

function isCopyCandidate(s) {
  if (!s || s.length < 2) return false;
  if (HAS_CHINESE.test(s)) return false;
  if (!/[a-zA-Z]/.test(s)) return false;
  if (SKIP_RE.test(s)) return false;
  if (TAILWIND_LIKE.test(s) && !s.includes(" ")) return false;
  if (s.length > 800) return false;
  return true;
}

function dedupe(items) {
  const seen = new Set();
  const out = [];
  for (const item of items) {
    const key = item.toLowerCase();
    if (seen.has(key)) continue;
    seen.add(key);
    out.push(item);
  }
  return out;
}

function extractFromSource(text) {
  const found = [];

  for (const m of text.matchAll(/"(?:[^"\\]|\\.)*"/g)) {
    const s = decodeJsString(m[0]);
    if (isCopyCandidate(s)) found.push(s);
  }

  for (const m of text.matchAll(/'(?:[^'\\]|\\.)*'/g)) {
    const s = decodeJsString(m[0]);
    if (isCopyCandidate(s)) found.push(s);
  }

  for (const m of text.matchAll(/>([^<>{}]+)</g)) {
    const chunk = m[1].replace(/\s+/g, " ").trim();
    if (isCopyCandidate(chunk)) found.push(chunk);
  }

  for (const m of text.matchAll(
    /(?:title|desc|description|label|eyebrow|subtitle|paragraph\d*|intro|closing|benefits|capabilities|primaryCtaText|secondaryCtaText|alt)\s*:\s*\n?\s*"([^"]+)"/gi
  )) {
    const s = m[1].trim();
    if (isCopyCandidate(s)) found.push(s);
  }

  return dedupe(found);
}

function walkTsx(dirRel) {
  const dir = path.join(SRC, dirRel);
  if (!fs.existsSync(dir)) return [];
  const files = [];
  const walk = (d) => {
    for (const ent of fs.readdirSync(d, { withFileTypes: true })) {
      const full = path.join(d, ent.name);
      if (ent.isDirectory()) walk(full);
      else if (ent.name.endsWith(".tsx") || ent.name.endsWith(".ts"))
        files.push(full);
    }
  };
  walk(dir);
  return files.sort();
}

function readFiles(paths) {
  const all = [];
  for (const p of paths) {
    if (fs.existsSync(p)) {
      all.push(...extractFromSource(fs.readFileSync(p, "utf8")));
    }
  }
  return dedupe(all);
}

function genericServiceCopy(slug) {
  const found = findService(slug);
  if (!found) return [];
  const { item, group } = found;
  const name = item.nameEn;
  return [
    "Services",
    name,
    group.titleEn,
    name,
    group.titleEn,
    `Professional manufacturing for ${name.toLowerCase()}. Custom specifications, quality materials, and reliable delivery for your brand or business.`,
    "Request a Quote",
    "View Quality Process",
    "Image placeholder (you can replace with real product photos later)",
  ];
}

function customizationSlugCopy(slug) {
  const item = CUSTOMIZATION_ITEMS.find((i) => i.slug === slug);
  if (!item) return [];
  const name = item.nameEn;
  return [
    "Home",
    "Customization",
    name,
    name,
    `Our ${name.toLowerCase()} services help you create unique, branded products that stand out in the market.`,
    "Get Started",
  ];
}

function servicesListCopy() {
  const lines = [
    "Our Services",
    "Comprehensive apparel manufacturing for every category. Browse by category or find your product below.",
  ];
  for (const group of SERVICE_GROUPS) {
    lines.push(group.titleEn);
    for (const item of group.items) lines.push(item.nameEn);
  }
  return lines;
}

function customizationListCopy() {
  const lines = [
    "Customization Services",
    "From printing to tech packs, we offer full customization solutions for your apparel brand.",
  ];
  for (const item of CUSTOMIZATION_ITEMS) lines.push(item.nameEn);
  return lines;
}

function extractAboutUs() {
  const defaults = fs.readFileSync(
    path.join(SRC, "lib", "aboutUsDefaults.ts"),
    "utf8"
  );
  const lines = extractFromSource(defaults);
  lines.push(...readFiles(walkTsx("components/about")));
  lines.push(
    ...readFiles(
      ["app/about-us/page.tsx", "app/about-us/AboutUsPageClient.tsx"].map((p) =>
        path.join(SRC, p)
      )
    )
  );
  return dedupe(lines);
}

function linesToParagraphs(lines) {
  const paras = [];
  for (const line of lines) {
    if (line.includes("\n")) {
      for (const part of line.split("\n")) {
        if (part.trim()) {
          paras.push(
            new Paragraph({ text: part.trim(), bullet: { level: 0 } })
          );
        }
      }
    } else {
      paras.push(new Paragraph({ children: [new TextRun(line)] }));
    }
  }
  return paras;
}

async function writeDocx(filename, title, route, lines) {
  const children = [
    new Paragraph({
      text: title,
      heading: HeadingLevel.TITLE,
    }),
    new Paragraph({
      children: [
        new TextRun({ text: "Route: ", bold: true }),
        new TextRun(route),
      ],
    }),
    new Paragraph({ text: "" }),
    new Paragraph({
      text: "Page copy (English)",
      heading: HeadingLevel.HEADING_1,
    }),
    ...linesToParagraphs(lines),
  ];

  const doc = new Document({ sections: [{ children }] });
  const buf = await Packer.toBuffer(doc);
  fs.writeFileSync(path.join(OUT_DIR, filename), buf);
}

function buildPages() {
  const pages = [];
  let idx = 1;

  const add = (label, route, lines, stem = null) => {
    const num = String(idx).padStart(2, "0");
    const fname = stem ? `${num}-${stem}.docx` : `${num}-${label}.docx`;
    pages.push({ idx, label, fname, route, lines });
    idx++;
  };

  add(
    "首页",
    "/",
    readFiles([
      path.join(SRC, "app", "page.tsx"),
      ...walkTsx("components/home"),
    ]),
    "首页"
  );

  add("服务列表", "/services", servicesListCopy(), "服务列表");

  for (const group of SERVICE_GROUPS) {
    for (const item of group.items) {
      const slug = item.slug;
      const rel = SERVICE_COMPONENT_DIRS[slug];
      const lines = rel
        ? readFiles(walkTsx(rel))
        : genericServiceCopy(slug);
      add(`服务-${item.nameEn}`, `/services/${slug}`, lines, `服务-${slug}`);
    }
  }

  add("定制列表", "/customization", customizationListCopy(), "定制列表");

  for (const item of CUSTOMIZATION_ITEMS) {
    add(
      `定制-${item.nameEn}`,
      `/customization/${item.slug}`,
      customizationSlugCopy(item.slug),
      `定制-${item.slug}`
    );
  }

  add("关于我们", "/about-us", extractAboutUs(), "关于我们");

  const blogLines = readFiles([
    path.join(SRC, "app", "blog", "page.tsx"),
    ...walkTsx("components/blog"),
  ]);
  const blogDetail = readFiles([
    path.join(SRC, "app", "blog", "[slug]", "page.tsx"),
    path.join(SRC, "components", "blog", "BlogPostBody.tsx"),
    path.join(SRC, "components", "blog", "BlogRelatedPosts.tsx"),
  ]);
  add("博客列表", "/blog", blogLines, "博客列表");
  add("博客文章模板", "/blog/[slug]", blogDetail, "博客-文章模板");

  add(
    "联系我们",
    "/contact-us",
    readFiles([path.join(SRC, "app", "contact-us", "page.tsx")]),
    "联系我们"
  );

  add(
    "质量与生产",
    "/quality",
    readFiles([path.join(SRC, "app", "quality", "page.tsx")]),
    "质量与生产"
  );

  add(
    "案例研究",
    "/company/case-studies",
    readFiles([
      path.join(SRC, "app", "company", "case-studies", "page.tsx"),
    ]),
    "案例研究"
  );

  add(
    "搜索",
    "/search",
    readFiles([
      path.join(SRC, "app", "search", "page.tsx"),
      path.join(SRC, "components", "search", "SearchForm.tsx"),
      path.join(SRC, "components", "search", "SearchResultsList.tsx"),
    ]),
    "搜索"
  );

  add(
    "隐私政策",
    "/privacy-policy",
    readFiles([path.join(SRC, "app", "privacy-policy", "page.tsx")]),
    "隐私政策"
  );

  add(
    "条款与条件",
    "/terms-and-conditions",
    readFiles([path.join(SRC, "app", "terms-and-conditions", "page.tsx")]),
    "条款与条件"
  );

  return pages;
}

async function main() {
  fs.mkdirSync(OUT_DIR, { recursive: true });

  const pages = buildPages();
  const manifest = [];

  for (const { idx, label, fname, route, lines } of pages) {
    const copy =
      lines.length > 0
        ? lines
        : ["(No extractable copy found in source for this page.)"];
    await writeDocx(fname, label, route, copy);
    manifest.push({
      index: idx,
      label,
      file: fname,
      route,
      lineCount: copy.length,
    });
    console.log(`  ${fname}  (${copy.length} lines)`);
  }

  fs.writeFileSync(
    path.join(OUT_DIR, "manifest.json"),
    JSON.stringify(manifest, null, 2),
    "utf8"
  );
  console.log(`\nDone: ${pages.length} documents -> ${OUT_DIR}`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: scripts/export-copy-to-docx.py
#!/usr/bin/env python3
"""Export public-site copy from suhucustom source into one .docx per page."""

from __future__ import annotations

import json
import re
import sys
from pathlib import Path

try:
    from docx import Document
    from docx.shared import Pt
except ImportError:
    print("Installing python-docx...", file=sys.stderr)
    import subprocess

    subprocess.check_call([sys.executable, "-m", "pip", "install", "python-docx", "-q"])
    from docx import Document
    from docx.shared import Pt

ROOT = Path(__file__).resolve().parents[1]
SRC = ROOT / "src"
OUT_DIR = ROOT / "suhucustom-文案导出"

# --- navigation (mirrors src/lib/navigation.ts) ---
SERVICE_GROUPS = [
    {
        "titleEn": "Tops & Activewear",
        "items": [
            {"slug": "t-shirts", "nameEn": "T-shirts"},
            {"slug": "hoodies-sweatshirts", "nameEn": "Hoodies & Sweatshirts"},
            {"slug": "activewear-athleisure", "nameEn": "Activewear & Athleisure"},
            {"slug": "gym-sportswear", "nameEn": "Gym & Sportswear"},
        ],
    },
    {
        "titleEn": "Bottoms & Underwear",
        "items": [
            {"slug": "leggings", "nameEn": "Leggings"},
            {"slug": "jeans-denim", "nameEn": "Jeans & Denim"},
            {"slug": "underwear-bras", "nameEn": "Underwear & Bras"},
            {"slug": "swimwear", "nameEn": "Swimwear"},
        ],
    },
    {
        "titleEn": "Accessories & Headwear",
        "items": [
            {"slug": "hats-headwear", "nameEn": "Hats & Headwear"},
            {"slug": "socks", "nameEn": "Socks"},
            {"slug": "neck-gaiters", "nameEn": "Neck Gaiters"},
            {"slug": "leather-goods", "nameEn": "Leather Goods"},
        ],
    },
    {
        "titleEn": "Specialized & Home",
        "items": [
            {"slug": "uniforms", "nameEn": "Uniforms"},
            {"slug": "baby-kids-clothing", "nameEn": "Baby & Kids Clothing"},
            {"slug": "towels", "nameEn": "Towels"},
            {"slug": "cushion-covers", "nameEn": "Cushion Covers"},
        ],
    },
]

CUSTOMIZATION_ITEMS = [
    {"slug": "printing", "nameEn": "Printing"},
    {"slug": "embroidery", "nameEn": "Embroidery"},
    {"slug": "private-label", "nameEn": "Private Label"},
    {"slug": "tech-pack-design", "nameEn": "Tech Pack Design"},
]

SERVICE_COMPONENT_DIRS: dict[str, str] = {
    "t-shirts": "components/services/tshirts",
    "hoodies-sweatshirts": "components/services/hoodies",
    "activewear-athleisure": "components/services/activewear",
    "gym-sportswear": "components/services/gym",
    "leggings": "components/services/leggings",
    "jeans-denim": "components/services/jeans",
    "socks": "components/services/socks",
    "neck-gaiters": "components/services/neck-gaiters",
    "leather-goods": "components/services/leather-goods",
    "towels": "components/services/towels",
    "cushion-covers": "components/services/cushion-covers",
}

GENERIC_SERVICE_SLUGS = {
    "underwear-bras",
    "swimwear",
    "hats-headwear",
    "uniforms",
    "baby-kids-clothing",
}

SKIP_STRING_RE = re.compile(
    r"(^/|https?://|@/|\./|#|\.tsx?$|\.svg|\.png|\.jpg|YOUR_|wa\.me|"
    r"placeholder|object-|flex-|grid-|max-w-|px-|py-|sm:|lg:|"
    r"aria-|lucide|currentColor|viewBox|xmlns|className|href=|src=)",
    re.I,
)

TAILWIND_LIKE = re.compile(r"^[a-z0-9]+(?:-[a-z0-9]+)+$")


def find_service(slug: str) -> tuple[dict, dict] | None:
    for group in SERVICE_GROUPS:
        for item in group["items"]:
            if item["slug"] == slug:
                return item, group
    return None


def has_chinese(text: str) -> bool:
    return bool(re.search(r"[\u4e00-\u9fff]", text))


def decode_js_string(raw: str) -> str:
    s = raw[1:-1]
    s = s.replace("\\n", "\n").replace('\\"', '"').replace("\\'", "'")
    s = re.sub(r"\s+", " ", s).strip()
    return s


def is_copy_candidate(s: str) -> bool:
    if not s or len(s) < 2:
        return False
    if has_chinese(s):
        return False
    if not re.search(r"[a-zA-Z]", s):
        return False
    if SKIP_STRING_RE.search(s):
        return False
    if TAILWIND_LIKE.match(s) and " " not in s:
        return False
    if len(s) > 800:
        return False
    if re.fullmatch(r"[\d\s\W]+", s):
        return False
    return True


def extract_from_source(text: str) -> list[str]:
    found: list[str] = []

    for m in re.finditer(r'"(?:[^"\\]|\\.)*"', text):
        s = decode_js_string(m.group(0))
        if is_copy_candidate(s):
            found.append(s)

    for m in re.finditer(r"'(?:[^'\\]|\\.)*'", text):
        s = decode_js_string(m.group(0))
        if is_copy_candidate(s):
            found.append(s)

    for m in re.finditer(r">([^<>{}]+)<", text):
        chunk = m.group(1).strip()
        chunk = re.sub(r"\s+", " ", chunk)
        if is_copy_candidate(chunk):
            found.append(chunk)

    for m in re.finditer(
        r"(?:title|desc|description|label|eyebrow|subtitle|paragraph\d*|intro|closing|"
        r"benefits|capabilities|primaryCtaText|secondaryCtaText|excerpt|placeholder|alt)"
        r'\s*:\s*"([^"]+)"',
        text,
        re.I,
    ):
        s = decode_js_string('"' + m.group(1) + '"')
        if is_copy_candidate(s):
            found.append(s)

    for m in re.finditer(
        r"(?:title|desc|description|label|eyebrow|subtitle|paragraph\d*|intro|closing|"
        r"benefits|capabilities|primaryCtaText|secondaryCtaText)\s*:\s*\n\s*\"([^\"]+)\"",
        text,
        re.I,
    ):
        s = m.group(1).strip()
        if is_copy_candidate(s):
            found.append(s)

    return dedupe_preserve_order(found)


def dedupe_preserve_order(items: list[str]) -> list[str]:
    seen: set[str] = set()
    out: list[str] = []
    for item in items:
        key = item.lower()
        if key in seen:
            continue
        seen.add(key)
        out.append(item)
    return out


def read_files(paths: list[Path]) -> list[str]:
    all_text: list[str] = []
    for p in paths:
        if p.is_file():
            all_text.extend(extract_from_source(p.read_text(encoding="utf-8", errors="replace")))
    return dedupe_preserve_order(all_text)


def glob_tsx(dir_rel: str) -> list[Path]:
    d = SRC / dir_rel
    if not d.exists():
        return []
    return sorted(d.glob("**/*.tsx"))


def extract_about_us() -> list[str]:
    defaults = (SRC / "lib" / "aboutUsDefaults.ts").read_text(encoding="utf-8", errors="replace")
    strings = extract_from_source(defaults)
    about_components = glob_tsx("components/about")
    strings.extend(read_files(about_components))
    app_files = [
        SRC / "app" / "about-us" / "page.tsx",
        SRC / "app" / "about-us" / "AboutUsPageClient.tsx",
    ]
    strings.extend(read_files([p for p in app_files if p.exists()]))
    return dedupe_preserve_order(strings)


def generic_service_copy(slug: str) -> list[str]:
    found = find_service(slug)
    if not found:
        return []
    item, group = found
    name = item["nameEn"]
    return [
        "Services",
        name,
        group["titleEn"],
        name,
        group["titleEn"],
        f"Professional manufacturing for {name.lower()}. Custom specifications, quality materials, and reliable delivery for your brand or business.",
        "Request a Quote",
        "View Quality Process",
        "Image placeholder (you can replace with real product photos later)",
    ]


def customization_slug_copy(slug: str) -> list[str]:
    item = next((i for i in CUSTOMIZATION_ITEMS if i["slug"] == slug), None)
    if not item:
        return []
    name = item["nameEn"]
    return [
        "Home",
        "Customization",
        name,
        name,
        f"Our {name.lower()} services help you create unique, branded products that stand out in the market.",
        "Get Started",
    ]


def services_list_copy() -> list[str]:
    lines = [
        "Our Services",
        "Comprehensive apparel manufacturing for every category. Browse by category or find your product below.",
    ]
    for group in SERVICE_GROUPS:
        lines.append(group["titleEn"])
        for item in group["items"]:
            lines.append(item["nameEn"])
    return lines


def customization_list_copy() -> list[str]:
    lines = [
        "Customization Services",
        "From printing to tech packs, we offer full customization solutions for your apparel brand.",
    ]
    for item in CUSTOMIZATION_ITEMS:
        lines.append(item["nameEn"])
    return lines


def write_docx(filename: str, title: str, route: str, lines: list[str]) -> None:
    doc = Document()
    style = doc.styles["Normal"]
    style.font.name = "Calibri"
    style.font.size = Pt(11)

    doc.add_heading(title, level=0)
    p = doc.add_paragraph()
    p.add_run("Route: ").bold = True
    p.add_run(route)

    doc.add_paragraph()
    doc.add_heading("Page copy (English)", level=1)

    for line in lines:
        if "\n" in line:
            for part in line.split("\n"):
                if part.strip():
                    doc.add_paragraph(part.strip(), style="List Bullet")
        else:
            doc.add_paragraph(line)

    path = OUT_DIR / filename
    doc.save(path)


# Page manifest: (index, chinese label, filename stem, route, copy lines)
def build_pages() -> list[tuple[int, str, str, str, list[str]]]:
    pages: list[tuple[int, str, str, str, list[str]]] = []
    idx = 1

    def add(label: str, route: str, lines: list[str], stem: str | None = None):
        nonlocal idx
        num = f"{idx:02d}"
        fname = f"{num}-{label}.docx" if stem is None else f"{num}-{stem}.docx"
        pages.append((idx, label, fname, route, lines))
        idx += 1

    home_files = [SRC / "app" / "page.tsx", *glob_tsx("components/home")]
    add("首页", "/", read_files(home_files), "首页")

    add("服务列表", "/services", services_list_copy(), "服务列表")

    for group in SERVICE_GROUPS:
        for item in group["items"]:
            slug = item["slug"]
            route = f"/services/{slug}"
            label = f"服务-{item['nameEn']}"
            if slug in SERVICE_COMPONENT_DIRS:
                rel = SERVICE_COMPONENT_DIRS[slug]
                lines = read_files(glob_tsx(rel))
            else:
                lines = generic_service_copy(slug)
            add(label, route, lines, f"服务-{slug}")

    add("定制列表", "/customization", customization_list_copy(), "定制列表")

    for item in CUSTOMIZATION_ITEMS:
        slug = item["slug"]
        add(
            f"定制-{item['nameEn']}",
            f"/customization/{slug}",
            customization_slug_copy(slug),
            f"定制-{slug}",
        )

    add("关于我们", "/about-us", extract_about_us(), "关于我们")

    blog_files = [
        SRC / "app" / "blog" / "page.tsx",
        *glob_tsx("components/blog"),
    ]
    blog_lines = read_files(blog_files)
    # Blog post detail shell (static UI only; article body is from CMS)
    blog_detail = read_files(
        [
            SRC / "app" / "blog" / "[slug]" / "page.tsx",
            SRC / "components" / "blog" / "BlogPostBody.tsx",
            SRC / "components" / "blog" / "BlogRelatedPosts.tsx",
        ]
    )
    add("博客列表", "/blog", blog_lines, "博客列表")
    add(
        "博客文章模板",
        "/blog/[slug]",
        blog_detail,
        "博客-文章模板",
    )

    add(
        "联系我们",
        "/contact-us",
        read_files([SRC / "app" / "contact-us" / "page.tsx"]),
        "联系我们",
    )

    add(
        "质量与生产",
        "/quality",
        read_files([SRC / "app" / "quality" / "page.tsx"]),
        "质量与生产",
    )

    add(
        "案例研究",
        "/company/case-studies",
        read_files([SRC / "app" / "company" / "case-studies" / "page.tsx"]),
        "案例研究",
    )

    search_files = [
        SRC / "app" / "search" / "page.tsx",
        SRC / "components" / "search" / "SearchForm.tsx",
        SRC / "components" / "search" / "SearchResultsList.tsx",
    ]
    add("搜索", "/search", read_files(search_files), "搜索")

    add(
        "隐私政策",
        "/privacy-policy",
        read_files([SRC / "app" / "privacy-policy" / "page.tsx"]),
        "隐私政策",
    )

    add(
        "条款与条件",
        "/terms-and-conditions",
        read_files([SRC / "app" / "terms-and-conditions" / "page.tsx"]),
        "条款与条件",
    )

    return pages


def main() -> None:
    OUT_DIR.mkdir(parents=True, exist_ok=True)
    pages = build_pages()
    manifest = []

    for num, label, fname, route, lines in pages:
        if not lines:
            lines = ["(No extractable copy found in source for this page.)"]
        write_docx(fname, label, route, lines)
        manifest.append({"index": num, "label": label, "file": fname, "route": route, "lineCount": len(lines)})
        print(f"  {fname}  ({len(lines)} lines)")

    (OUT_DIR / "manifest.json").write_text(
        json.dumps(manifest, ensure_ascii=False, indent=2), encoding="utf-8"
    )
    print(f"\nDone: {len(pages)} documents -> {OUT_DIR}")


if __name__ == "__main__":
    main()
